### DISCUSSION TITLE
Gzip compression seemingly not working out of the box

### DISCUSSION THREAD

**Discussion by durran-kelly** [dropwizard/dropwizard#4782] [General]
Hello, 

I am experiencing issues with getting my api response to be compressed. I have been making my requests in Postman and have confirmed that it isn't automatically decoding the gzip responses nor striping the Content-Encoding header. I was able to test this by making a request to http://httpbin.org/gzip. I also used a quick start SpringBoot project and turned on gzip encoding. I was able to see that my response was encoded properly and Content-Encoding: gzip was also present in the header.

I also turned on logging and see this response:
DEBUG [2021-03-24 04:29:31,166] org.eclipse.jetty.server.handler.gzip.GzipHandler: GzipHandler@a2ddf26{STARTED,min=256,inflate=8192} handle Request(POST //localhost:8080/v1/volumes)@7d5ea0d4 in null
DEBUG [2021-03-24 04:29:31,167] org.eclipse.jetty.server.handler.gzip.GzipHandler: GzipHandler@a2ddf26{STARTED,min=256,inflate=8192} excluded by method Request(POST //localhost:8080/v1/volumes)@7d5ea0d4

The "in null" part is a little concerning.

I have tried to explicitly enable gzipping through the config with no success (despite it being enabled by default).

Can someone please tell me what I am doing wrong? Thanks!!

**Comment by joschi** [comment]
@durran-kelly Which version of Dropwizard are you using? What's the complete configuration of your Dropwizard application?

Please post the exact HTTP requests your sending and their responses (incl. all headers of request and response).

**Comment by durran-kelly** [comment]
@joschi Thanks for the response.

Dropwizard version: 2.0.20

config.yml: 
```yaml
logging:
  level: INFO
  loggers:
    org.eclipse.jetty.server.handler.gzip: DEBUG
```

Request made with a JSON body:
```
POST http://localhost:8080/v1/volumes
```
Headers: 
```
Content-Type: application/json
Content-Length: <calculated when request is sent>
Host: <calculated when request is sent>
User-Agent: PostmanRuntime/7.26.8
Accept: */*
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
```

Response includes response body 
Header: 
```
 Date: Wed, 24 Mar 2021 04:29:31 GMT
 Content-Type: application/json
 Transfer-Encoding: chunked 
```

**Comment by joschi** [comment]
@durran-kelly The Jetty `GzipHandler` has some (configurable) restrictions and will by default only compress responses to `GET` requests with a size larger than 256 bytes.

See https://www.dropwizard.io/en/release-2.0.x/manual/configuration.html#gzip for the configuration settings in Dropwizard and https://www.javadoc.io/doc/org.eclipse.jetty/jetty-server/9.4.38.v20210224/org/eclipse/jetty/server/handler/gzip/GzipHandler.html for the implementation details of the Jetty `GzipHandler` class.

**Comment by durran-kelly** [comment]
@joschi Thanks for the response and pointing out the fact its GET by default (totally missed that in the documentation...) Sorry for the trouble!

**Comment by joschi** [comment]
@durran-kelly All good. Glad we could answer your questions and that it's working for you now. 😄
